components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt

```diff
@@ -199,6 +205,107 @@ class GitpodManager : Disposable {
         }
     }
 
+    private val portsObserveJob = GlobalScope.launch {
+        if (application.isHeadlessEnvironment) {
+            return@launch
+        }
+
+        val ignorePorts = listOf(5990, 6942)
+        val portsStatus = hashMapOf<Int, PortsStatus>()
+
+        val status = StatusServiceGrpc.newStub(supervisorChannel)
+        //while (isActive) {
```

Comment:
@akosyakov if I keep this `while` - it creates for each loop a new subscription to supervisor and soon it starts throwing errors for too many concurrent subscriptions. Are we sure I need it?

Reply:
you should call `f.await()` as here https://github.com/gitpod-io/gitpod/blob/641adda863bbe7c904f34f8d94b6e866f89d9feb/components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt#L186

Reply:
Please make sure to align with how we listen to notifications 🙏 it is done right there